libraries/AP_BattMonitor/examples/AP_BattMonitor_test/AP_BattMonitor_test.cpp

```diff
@@ -18,7 +19,7 @@ void setup() {
     hal.console->printf("Battery monitor library test\n");
 
     // set battery monitor to smbus
-    battery_mon.set_monitoring(0, AP_BattMonitor::BattMonitor_TYPE_SOLO);
+    AP_Param::set_by_name("BATT_MONITOR",  AP_BattMonitor_Params::BattMonitor_TYPE_SOLO);
```

Comment:
I'm not following this comment at all, could you explain a bit more what you mean?

First of all the prefix is "BATT" from the vehicle (look at plane and sub).

Second I have not gone out of my way to test the examples (I refuse to run them as root, which I'm given to understand means I can't run them) however the call was taken directly from sub 
 https://github.com/ArduPilot/ardupilot/blob/master/ArduSub/system.cpp#L42 and the BATT_MONITOR param will exist.

Side not now that we're talking about it. I somewhat doubt anyone has convincingly run that example, because you can't run SITL/similair with a SMBus driver as there is no fake I2C bus. (I'm not convinced the entire example holds any value but thats a different discussion).

Reply:
> First of all the prefix is "BATT" from the vehicle (look at plane and sub).

But there is no vehicle in the example... Or better yet, the example is the vehicle, but it has no parameter table.
I'm not sure how else to explain it. When you build an example you aren't building Plane, Copter, Sub, Rover or Tracker, you are building just the example. As you know, the library parameter prefixes are defined in the param tables in the vehicles - there isn't one here, so where does the *BATT_* come from?

> Second I have not gone out of my way to test the examples (I refuse to run them as root, which I'm given to understand means I can't run them) however the call was taken directly from sub
https://github.com/ArduPilot/ardupilot/blob/master/ArduSub/system.cpp#L42 and the BATT_MONITOR param will exist.

Again, you are talking about vehicles that have parameter tables including those parameters. If you go to Sub and change the GND_ to something else (or remove it), then that code you linked to will stop working.

> Side not now that we're talking about it. I somewhat doubt anyone has convincingly run that example, because you can't run SITL/similair with a SMBus driver as there is no fake I2C bus. (I'm not convinced the entire example holds any value but thats a different discussion). (If you set the driver to use a SMBus one in SITL then restart you get a segfault)

You know you can run the examples on a real board right? So, you can run just the example on a Solo for example.